src/common/k8s-api/kube-object.store.ts

```diff
@@ -7,32 +7,32 @@ import type { ClusterContext } from "./cluster-context";
 
 import { action, computed, makeObservable, observable, reaction, when } from "mobx";
 import type { Disposer } from "../utils";
-import { autoBind, noop, rejectPromiseBy } from "../utils";
-import type { KubeObject } from "./kube-object";
+import { autoBind, includes, isRequestError, noop, rejectPromiseBy } from "../utils";
+import type { KubeJsonApiDataFor, KubeObject } from "./kube-object";
 import { KubeStatus } from "./kube-object";
 import type { IKubeWatchEvent } from "./kube-watch-event";
 import { ItemStore } from "../item.store";
-import type { IKubeApiQueryParams, KubeApi } from "./kube-api";
-import { ensureObjectSelfLink } from "./kube-api";
+import type { KubeApiQueryParams, KubeApi, KubeApiWatchCallback } from "./kube-api";
 import { parseKubeApi } from "./kube-api-parse";
-import type { KubeJsonApiData } from "./kube-json-api";
 import type { RequestInit } from "node-fetch";
-
-// BUG: https://github.com/mysticatea/abort-controller/pull/22
-// eslint-disable-next-line import/no-named-as-default
 import AbortController from "abort-controller";
 import type { Patch } from "rfc6902";
 import logger from "../logger";
+import assert from "assert";
+import type { PartialDeep } from "type-fest";
+import { entries } from "../utils/objects";
+
+export type OnLoadFailure = (error: unknown) => void;
 
 export interface KubeObjectStoreLoadingParams {
   namespaces: string[];
-  reqInit?: RequestInit;
+  reqInit?: RequestInit | undefined;
```

Comment:
Why do we have to add `| undefined`? Isn't `?` enough?

Reply:
I think this is a remnant of experimenting with turning on https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes (which I did not end up doing). So this isn't actually required to be done since the compiler will add the `| undefined` automatically.